### DISCUSSION TITLE
Consider Titanfall 2-like wallrunning/hopping

### DISCUSSION THREAD

**Discussion by poperigby** [redeclipse/base#1171] [Ideas and Feature Requests]
I think Red Eclipse's parkour and Titanfall 2's parkour are both pretty similar, because they both have wall running and jumping. The main thing that makes them feel different is that TF2's parkour feels more intuitive to use. It's more intuitive because you don't have to hold down a button to wall run or jump up a wall, you just jump on to a  wall and you start running along it or you start hopping up it. Warframe does something similar with it's movement. I think this would be a great idea, because it doesn't take away from the complexity of the movement, but also makes it more accessible for newer players, and like I said before, makes it feel much more intuitive. [This](https://www.youtube.com/watch?v=n1mbGmOZWcU) video does a pretty good job illustrating how Titanfall 2's wall running works. I really don't think it would be that big of a change, but it would make the game feel and play a lot better. Also, [sliding](https://www.youtube.com/watch?v=AYu15tvaUeU) would be pretty cool as well, but that's less important I think. Although it would fit nicely.

**Comment by qreeves** [comment]
Titanfall 2's maps aren't small arenas where you're constantly bumping into walls and other things (they also do some hacky things like making certain surfaces non-parkourable, which RE tries to avoid). The special key prevents accidental wall runs/climbs and performs the melee kick move in mid-air to damage actors (and allowing you to bounce off their head). You can slide in RE by holding crouch as you land on the ground, and while doing so you perform another sort of melee kick (if you manage to run into an actor with it).

**Comment by poperigby** [comment]
There are actually a lot of super tight spaces in TF2 (usually when you're inside a building), and I never wall run when I don't want to. It's very hard to accidentally stick onto the wall.

The kick could also be bound to the 'q' if there is no need for the special key.

For the sliding, there isn't a very good visually cue so you realize you're doing it. Maybe show your feet in a sliding position?

**Comment by byronsharman** [comment]
My personal take on this is that having to put Q in your muscle memory is OK; it is part of learning the game and, while I don't remember much of learning RE, isn't the most difficult part of learning parkour.

**Comment by poperigby** [comment]
Putting it in my muscle memory isn't that difficult, it just doesn't feel very intuitive and doesn't make a lot of sense to me.

**Comment by SniperGoth** [comment]
I've played Titanfall 2 quite a lot last year, and yes, i can confirm it is very intuitive to just jump into a wall and wallrun automatically on there.

As for the proposal, we *could * have automatic wallrun as a secondary method of doing it, as "wallrunstyle 1" or something, so people could choose between both in the options menu.

As for the sliding, i also agree it isn't intuitive to jump to slide (in general you just run and crouch to activate it on games with more advanced movement), and i feel just running (for 1000 time) and crouch could do the same for it as an alternative.

It's up to Qreeves to decide, but take these on consideration as alternatives to the classic method of execution.

**Comment by poperigby** [comment]
> As for the proposal, we *could * have automatic wallrun as a secondary method of doing it, as "wallrunstyle 1" or something, so people could choose between both in the options menu.

That would work well.

> As for the sliding, i also agree it isn't intuitive to jump to slide (in general you just run and crouch to activate it on games with more advanced movement), and i feel just running (for 1000 time) and crouch could do the same for it as an alternative.

Those were also my thoughts. Just crouching while running would be easier than having to jump first.

**Comment by qreeves** [comment]
You don't think I haven't tried this in the 10 years we've had parkour?

It isn't about being intuitive, it's about being deliberate while providing a limiting factor. Any sort of automatic method for doing these moves would remove the ability to perform them while not facing the wall, and in turn remove the ability to aim in a totally different direction while you run along them. Both wall running and sliding is too powerful in RE for it to be easy to perform, and people who've played the game for more than a few days should know this already.

The physics in Titanfall are very different to what we have in RE, and these kinds of analogies are spurious at best. It's like looking at a jigsaw puzzle piece and saying it fits in every puzzle that has the same colour somewhere on it. It's easy to make this assumption when you look at the single piece, but becomes obviously wrong when you look at the bigger picture.

I try to be nice, but people never like the answer and always argue. I should learn to just say no.

**Comment by cloudziq** [comment]
qreeves is right on this,  any sort of auto-moves will introduce more chaos and reduce 'things to master'.  Simplicity is great, but controling your avatar is important too.  If RE will become too easy, it will be more boring over the time you play.

**Comment by SniperGoth** [comment]
Oh well, more intuitive options to do things wouldn't hurt (like the running into slide) imo, but i get your point of view.
I would still keep this in mind, mostly because 2.0 is a soft reboot of the game in general.
That's not up to me to decide tho.

**Comment by poperigby** [comment]
Sorry. I wasn't trying to be annoying.

**Comment by qreeves** [comment]
I'll look at the slide when we start working on redesigning it (along with adding back a dash mechanic).

With regard to automatic wall running/climbing, put it this way: when I tested this out it became very difficult to complete a lot of race maps due to the cramped space and the need to wall run at specific points on the wall. Eg. you're at a point where you need to drop down through a tunnel and then wall run at the end of the drop (like the drops in `testchamber`), if you so much as bumped the wall you'd waste a wall run and precious time.

**Comment by poperigby** [comment]
Is the game able to detect when your right next to a wall?

**Comment by SniperGoth** [comment]
Type this on your config file, under my games/Red Eclipse and you get a automatic wallrun of sorts.

bind "SPACE"     [ jump; special ] 

Done.
But keep in mind that this bind makes it so that when you press space, it automatically also presses the "special" key, which is the parkour movement.
So, you hold space to enter the walls/climb, if you keep holding it it will continue doing the wallruns/climb, and if you press it again you jump.
You may waste a bit more moves than normal sometimes since you are technically doing 2 impulse moves at the same time, though.
You have more intuitive control now, but you lose just a little bit of manual control, so take that into consideration.

I'll put a video later showing this in action.

**Comment by poperigby** [comment]
Would it be difficult to make a CVAR or user settings for auto-wall running?

**Comment by acerspyro** [comment]
I don't believe most users want that, since as we've explained, this goes against the game's mechanic and breaks parkour.

**Comment by poperigby** [comment]
Maybe there could be another way to make the special key easier to use, with out breaking the parkour, binding it to a mouse button would be a lot easier, but I'm pretty sure those are already taken up.

**Comment by acerspyro** [comment]
If you have a 5 button mouse, you can bind buttons 4 and 5 (usually side buttons) to your parkour ability.

The Q key is easily reachable during gameplay using your ring finger and is only necessary to kick and get on walls, the player stays on walls automatically.

**Comment by poperigby** [comment]
Oh. They stay on walls automatically? Well, that's a lot easier than having to hold it down.

**Comment by SniperGoth** [comment]
As promised : https://youtu.be/qqe3LPesIwQ

Auto parkour with hand view to go along... with a custom shotgun. Personal stuff, ignore that lol

**Comment by poperigby** [comment]
Wow! That looks great! That's exactly what I was thinking of. How did you do that so fast? The shotgun also looks super cool. Love the spin cock.

**Comment by SniperGoth** [comment]
It's explained up there ^^ 



> Type this on your config file, under my games/Red Eclipse and you get a automatic wallrun of sorts.
> bind "SPACE"     [ jump; special ]
> Done.
> But keep in mind that this bind makes it so that when you press space, it automatically also presses the "special" key, which is the parkour movement.
> So, you hold space to enter the walls/climb, if you keep holding it it will continue doing the wallruns/climb, and if you press it again you jump.
> You may waste a bit more moves than normal sometimes since you are technically doing 2 impulse moves at the same time, though.
> You have more intuitive control now, but you lose just a little bit of manual control, so take that into consideration.

 You can try it yourself using that bind.

**Comment by poperigby** [comment]
Oh, right. I wonder if there's a way to make it work better, with out messing anything up.

**Comment by byronsharman** [comment]
> Maybe there could be another way to make the special key easier to use

As Acerspyro pointed out, it is very easy to use if you only spend a little time learning when it is pressed in order to put it in your muscle memory. RE is supposed to be a fast-paced and advanced game; I'm not sure a drastic change merely for the new players is a good idea.
